### ISSUE TITLE
Python version compatiblity - docs ambiguity

### ISSUE THREAD

**Issue by ewjoachim** [hadialqattan/pycln#139]
**Describe the issue**

There's an info I'd like to find in the docs, and I wasn't able to.
The documentation (and README) specify that the tool can run on Python 3.6 to 3.10. What it doesn't say is if it is also able to operate on code that is running with 3.6 to 3.10, and more importantly, if the version the tool runs on is required to be the same version as the version the codebase we use the tool on.

In other words, are there places in the code where the tool needs to figure out if it's working on a py3.6 or a py3.10 python file, and it uses sys.version to know ?

Other linting tools such as black and isort have a `--python-version` (or equivalent) flag to disambiguate. On the other hand, flake8 has a note in the docs saying it must run on the same version it operates on.

When using pre-commit, I want to know if I need to put `language_version: python3.x` or if I can just let it select the most recent python version.

**Comment by hadialqattan** [comment]
Hi @ewjoachim, this is a good question. The short answer is that the most recent version of Python will always work with all possible code bases while some problems will appear when you use a version to run Pycln older than the code base's one.

In case you want to know the exact behavior of Pycln regarding using a different version to run Pycln other than the code base's one, Pycln uses `sys.version_info` in [pycln/utils/scan.py](https://github.com/hadialqattan/pycln/blob/master/pycln/utils/scan.py) in only three different places:

1) For class validation (because py38< is missing some needed nodes' location properties): https://github.com/hadialqattan/pycln/blob/dc035b2e39c6e127d64377c2db6df6dadfab12d7/pycln/utils/scan.py#L156
_NOTE: this use case is only related to the Python version used to run Pycln, so you can use whatever version with whatever code base._

2) For taking [py3.8 type comments](https://www.python.org/dev/peps/pep-0526/) into account: https://github.com/hadialqattan/pycln/blob/dc035b2e39c6e127d64377c2db6df6dadfab12d7/pycln/utils/scan.py#L902-L905
_NOTE: using a version less than 3.8 with a code base containing a type comment will make Pycln ignore those type comments._

3) An `ast` module breaking change related check: https://github.com/hadialqattan/pycln/blob/dc035b2e39c6e127d64377c2db6df6dadfab12d7/pycln/utils/scan.py#L239-L240
_NOTE: this use case is only related to the Python version used to run Pycln, so you can use whatever version with whatever code base._

In conclusion, using different versions won't affect Pycln except for [py3.8 type comments](https://www.python.org/dev/peps/pep-0526/) case, also, when a new syntax is introduced in a newer version like the match statement in py3.10 (you should use py3.10 > for any code base that has match statements otherwise Pycln will not consider any import usage that's related to the match statement and will output a non-crushing SyntaxError to stderr).

<br>
<br>

Thanks for supporting Pycln by giving this feedback, I'm happy to see you around.

**Comment by hadialqattan** [comment]
btw, @ewjoachim do you think having something like Black/Isort's `--python-version` in Pycln is needed?

_(I'm asking because I doubt that in our case we could/should add something like that)._

**Comment by hadialqattan** [comment]
After telling your thoughts, at least I'm gonna add a note in the docs about that.

**Comment by ewjoachim** [comment]
Yes, I'm not worried "per se" about specific issue, I'd say adding a note in the docs could be largely enough. Something along the lines of "Make sure the Python version you run pycln with is the same or more recent than the python version your codebase targets".

I'd say it's a safe bet at least for the next few years to consider that after `async / await` becoming keywords, we may not have a lot of breaking changes where valid 3.x code would parse as different AST from 3.y (x<y).

So, all good for me :)

(And in case you didn't know, even if I agree that it's probably not needed in our case, `ast.parse()` can accept a version tuple to use that version's grammar)

**Comment by hadialqattan** [comment]
A note has been added to the [usage section](https://hadialqattan.github.io/pycln/#/?id=usage). Thanks for your participation.
